### PR TITLE
Move denoiser implementations out of `functional` sub-module

### PR DESCRIPTION
**Pull request by bwohlberg**
Move denoiser implementations out of `functional` sub-module. Partially addresses #188.

### CODE REVIEW

File: scico/denoiser.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        if variant[0] == "6":
+            nlayer = 6
+        else:
+            nlayer = 17
+        model = DnCNNNet(depth=nlayer, channels=1, num_filters=64, dtype=np.float32)
+        variables = load_weights(_flax_data_path("dncnn%s.npz" % variant))
+        super().__init__(model, variables)
+
+    def __call__(self, x: JaxArray) -> JaxArray:
+        r"""Apply DnCNN denoiser.
```
Comment (bwohlberg) [inline scico/denoiser.py:122]: @crstngc: Can the code here be re-factored so that this method can make use of `FlaxMap.__call__` rather than completely replacing it? If not, is there any real role for the intermediate `FlaxMap` class?

Comment (crstngc) [reply scico/denoiser.py:122]: The denoiser call is swapping the channel axis with the batch axis to be able to apply a grayscale trained model also to color images. `FlaxMap` does not do that, it directly applies a Flax model. Are you suggesting to always apply Flax models using axes swap?

File: scico/test/test_denoiser.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+
+class TestDnCNN:
+    def setup(self):
+        key = None
+        self.x_sngchn, key = randn((32, 33), key=key, dtype=np.float32)
+        self.x_mltchn, key = randn((33, 34, 5), key=key, dtype=np.float32)
+        self.dncnn = DnCNN()
+
+    def test_sngchn(self):
```
Comment (Michael-T-McCann) [inline scico/test/test_denoiser.py:60]: `test_single_channel`?

File: scico/test/test_denoiser.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        no_jit = self.dncnn(self.x_sngchn)
+        jitted = jax.jit(self.dncnn)(self.x_sngchn)
+        np.testing.assert_allclose(no_jit, jitted, rtol=1e-3)
+        assert no_jit.dtype == np.float32
+        assert jitted.dtype == np.float32
+
+    def test_mltchn(self):
```
Comment (Michael-T-McCann) [inline scico/test/test_denoiser.py:67]: `test_multiple_channels`?

**Review by Michael-T-McCann — APPROVED** [review]
Looks much cleaner.

File: scico/test/test_denoiser.py (hunk 2 above)
Comment (bwohlberg) [reply scico/test/test_denoiser.py]: Done.

File: scico/test/test_denoiser.py (hunk 3 above)
Comment (bwohlberg) [reply scico/test/test_denoiser.py]: Done.
